src/protocol/resp/src/request/badd.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use super::*;
+use std::io::{Error, ErrorKind};
+use std::sync::Arc;
+
+type ArcByteSlice = Arc<Box<[u8]>>;
+type ArcKeyValuePair = (ArcByteSlice, ArcByteSlice);
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct BAddRequest {
```

Comment:
Since `badd` command is not a upstream redis command, it would be good to document what this is. Adding some rustdoc for the struct would be helpful